pkg/model/labels.go

```diff
@@ -219,6 +219,52 @@ func (ls Labels) Delete(name string) Labels {
 	return ls
 }
 
+func (ls Labels) Subtract(labels Labels) Labels {
+	var i, j, k int
+	for i < len(ls) && j < len(labels) {
+		cmp := CompareLabelPairs2(ls[i], labels[j])
+		switch {
+		case cmp == 0:
+			i++
+			j++
+		case cmp < 0:
+			if i != k {
+				ls[k] = ls[i]
+			}
+			k++
+			i++
+		default:
+			j++
+		}
+	}
+	for i < len(ls) {
+		if i != k {
+			ls[k] = ls[i]
+		}
+		k++
+		i++
+	}
+	return ls[:k]
+}
+
+func (ls Labels) Intersect(labels Labels) Labels {
+	var i, j, k int
+	for i < len(ls) && j < len(labels) {
+		cmp := CompareLabelPairs2(ls[i], labels[j])
+		if cmp == 0 {
+			ls[k] = ls[i]
+			k++
+			i++
+			j++
+		} else if cmp < 0 {
+			i++
+		} else {
+			j++
+		}
+	}
+	return ls[:k]
+}
```

Comment:
This is one way to distinguish "sample labels" from "series labels": given a set of label sets, their intersection represents the "series labels".

This holds true in 99.9% of cases. If labels are set dynamically during program execution, some stack traces or samples should fall outside the scope where those labels can be applied, determining the intersection. In other cases – such as when we add static labels to all samples via the profiler – we treat the labels shared by all samples as "series labels".

This approach allows us to reconstruct the original pprof labels after a pprof split.